### ISSUE TITLE
Windows build failing

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#2945]
Seems like Win CI has issues again.. :roll_eyes: 

@mloskot do you maybe have a few mins? Today and tmrw I'm pretty caught up in project work, I can do it on friday the latest.

ref: https://dev.azure.com/valhalla1/valhalla/_build/results?buildId=8305&view=logs&j=8c5a3053-cc94-5e24-da52-c43aabf65e0e&t=de4ba696-e3b4-5027-1054-4f8c14f9b343

**Comment by mloskot** [comment]
Trying a fix in https://github.com/valhalla/valhalla/pull/2947

**Comment by mloskot** [comment]
It seems the `key: 'build | "$(Build.SourceBranch)" | "msvc-v16.9.0"'` is unfortunate as it correlates with existing cache for some reason:

https://dev.azure.com/valhalla1/valhalla/_build/results?buildId=8341&view=logs&j=8c5a3053-cc94-5e24-da52-c43aabf65e0e&t=a1bc7e75-1dfc-5d62-965d-76d879c54f66

```
Resolving key:
 - build                  [string]
 - "refs/pull/2947/merge" [string]
 - "msvc-v16.9.0"         [string]
Resolved to: build|"refs/pull/2947/merge"|"msvc-v16.9.0"
Resolving restore key:
 - build                  [string]
 - "refs/pull/2947/merge" [string]
Resolved to: build|"refs/pull/2947/merge"|**
Resolving restore key:
 - build [string]
Resolved to: build|**
ApplicationInsightsTelemetrySender will correlate events with X-TFS-Session c0e3c081-b6b6-4ee0-a878-0ecfbefb2767
Getting a pipeline cache artifact with one of the following fingerprints:
Fingerprint: `build|"refs/pull/2947/merge"|"msvc-v16.9.0"`
Fingerprint: `build|"refs/pull/2947/merge"|**`
Fingerprint: `build|**`
There is a cache hit: `build|"refs/heads/master"`
Used scope: 1;fe02f257-eab6-47d1-a6e7-6ddabf0b7678;refs/heads/master;valhalla/valhalla
Entry found at fingerprint: `build|"refs/heads/master"`
```
